src/codegen/mod.rs

```diff
@@ -1612,10 +1527,17 @@ impl CodeGenerator for CompInfo {
                 fields.push(padding_field);
             }
 
-            if let Some(align_field) =
-                layout.and_then(|layout| struct_layout.align_struct(layout))
-            {
-                fields.push(align_field);
+            if let Some(layout) = layout {
+                if struct_layout.requires_explicit_align(layout) {
+                    if layout.align == 1 {
+                        packed = true;
+                    } else {
+                        let ty = helpers::blob(Layout::new(0, layout.align));
+                        fields.push(quote! {
+                            pub __bindgen_align: #ty ,
```

Comment:
This will only work to *increase* alignment, right? So what if we are decreasing alignment (but not all the way to `1`)? I guess that requires `#pragma packed` or `__attribute__((aligned(..)))` which we don't fully support yet.

Reply:
Yep. If one would try to decrease alignment it would lead to layout test failure... just as it is now.